showdown/engine/special_effects/abilities/on_switch_in.py

```diff
@@ -6,7 +6,9 @@ def sandstream(state, attacking_side, attacking_pokemon, defending_side, defendi
         return [(
             constants.MUTATOR_WEATHER_START,
             constants.SAND,
-            state.weather
+            4,
+            state.weather,
+            state.remaining_weather_turns
         )]
     return None
 
```

Comment:
Each of these weather ability checks needs to now check for the existence of an extension item. If it exists the number would need to increase.